### PR TITLE
feat: type-safety for configurable endpoints (3399)

### LINKED ISSUE #3399: Add type-safe OCC endpoint configuration
In order to configure the endpoints for OCC, the dev needs to know the endpoint keys. These keys are currently any string, and the dev can only look at source code to understand the keys that must be used. 

Since OCC is fairly static, we can add the different keys in the model, and leave it open to add any strings. This small improvement will improve dev experience and saves a lot of time.

### PR DESCRIPTION
**Pull request by LTiger14**
Closes #3399

### CODE REVIEW

File: projects/core/src/occ/occ-models/occ-endpoints.model.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
+export interface OccEndpoints {
+  /**
+   * Get product details
+   *
+   * @member {string} [product]
+   */
+  product?: string;
+  /**
+   * Get reviews for a product
```
Comment (tobi-or-not-tobi) [inline projects/core/src/occ/occ-models/occ-endpoints.model.ts]: Get a list of product reviews

File: projects/core/src/occ/occ-models/occ-endpoints.model.ts [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+   *
+   * @member {string} [productReviews]
+   */
+  productReviews?: string;
+  /**
+   * Get a product reference
```
Comment (tobi-or-not-tobi) [inline projects/core/src/occ/occ-models/occ-endpoints.model.ts]: Get a list of product references

File: projects/core/src/occ/occ-models/occ-endpoints.model.ts [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+   */
+  productSearch?: string;
+  /**
+   * Get a list of available suggestions
+   *
+   * @member {string} [productSuggestings]
+   */
+  productSuggestions?: string;
+  /**
+   * Get component data
```
Comment (tobi-or-not-tobi) [inline projects/core/src/occ/occ-models/occ-endpoints.model.ts]: Get CMS component details

File: projects/core/src/occ/occ-models/occ-endpoints.model.ts [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+   *
+   * @member {string} [component]
+   */
+  component?: string;
+  /**
+   * Find component data by id(s)
```
Comment (tobi-or-not-tobi) [inline projects/core/src/occ/occ-models/occ-endpoints.model.ts]: Get a list of CMS component details

File: projects/core/src/occ/occ-models/occ-endpoints.model.ts [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+   *
+   * @member {string} [components]
+   */
+  components?: string;
+  /**
+   * Get page data with list of cms content slots
```
Comment (tobi-or-not-tobi) [inline projects/core/src/occ/occ-models/occ-endpoints.model.ts]: Do we even need to support this one? It's the same as `page`, but homepage is return if there's no page uid specified.

Comment (LTiger14) [reply projects/core/src/occ/occ-models/occ-endpoints.model.ts]: All of the endpoints listed are already in use. This is just a list of what is already there.

**Comment by LTiger14** [comment]
Closing PR to open a new one to 1.1 branch

**Pull request closed without merging**